home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/UserApiServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.softserveinc.ita.homeproject.application.apiservice;
+
+import com.softserveinc.ita.homeproject.api.NotFoundException;
+import com.softserveinc.ita.homeproject.api.UsersApiService;
+import com.softserveinc.ita.homeproject.model.CreateUser;
+import com.softserveinc.ita.homeproject.model.UpdateUser;
+import com.softserveinc.ita.homeproject.service.impl.UserServiceImpl;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+@Service
+@RequiredArgsConstructor
+public class UserApiServiceImpl implements UsersApiService {
+
+    private final UserServiceImpl userServiceImpl;
+
+    @PreAuthorize("hasAuthority('CREATE_USER_PERMISSION')")
+    @Override
+    public Response createUser(CreateUser createUser, SecurityContext securityContext) throws NotFoundException {
+        return Response.ok().entity(userServiceImpl.createUser(createUser)).build();
+    }
+
+    @PreAuthorize("hasAuthority('UPDATE_USER_PERMISSION')")
+    @Override
+    public Response updateUser(Integer id, UpdateUser updateUser, SecurityContext securityContext) throws NotFoundException {
```

Comment:
Do we really use `Integer` for id?

Reply:
Generator generates Integer value, I will look into it and see if it's possible to change it

Reply:
I will fix it